src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -281,6 +285,24 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 		column->varno = OUTER_VAR;
 	}
 
+	allTargetList = copyObject(masterQuery->targetList);
```

Comment:
Can we put this block into a function `HasDistinctAggregate()`?

Reply:
done